sdks/python/hatchet_sdk/worker/runner/utils/capture_logs.py

```diff
@@ -81,23 +108,78 @@ def __init__(self, event_client: EventClient):
         self.q = asyncio.Queue[LogRecord | STOP_LOOP_TYPE](
             maxsize=event_client.client_config.log_queue_size
         )
+        self.buffer = LogBuffer()
+
+        ## flag to see if the engine supports bulk log push
+        ## defaults to true, but reset to false if we get an
+        ## `UNIMPLEMENTED` or `NOT_FOUND` error back
+        self.bulk_log_push_enabled = True
+
+    async def flush(self) -> None:
+        ## IMPORTANT: Need a deepcopy here to act as a snapshot of the
+        ## current buffer, since new logs might be added while we're processing
+        ## the current buffer
+        records = copy.deepcopy(self.buffer.records)
+
+        requests = [
+            BulkLogPushRequest(
+                message=r.message,
+                task_run_external_id=r.task_run_external_id,
+                level=r.level,
+            )
+            for r in records
+        ]
+
+        try:
+            await asyncio.to_thread(self.event_client.bulk_log, requests)
+            self.buffer.records -= records
+        except grpc.RpcError as e:
+            if e.code() in [grpc.StatusCode.UNIMPLEMENTED, grpc.StatusCode.NOT_FOUND]:
+                self.bulk_log_push_enabled = False
```

Comment:
if we get one of these codes back, it means the engine was an old version that didn't have bulk log push